src/main/java/client/ClientHandler.java

```diff
@@ -1,29 +1,24 @@
 package client;
 
-import codec.Session;
+import session.Session;
 import message.Message;
 import message.MessageType;
 import server.Viewer;
 
 import java.io.IOException;
+import java.util.Objects;
 
 public class ClientHandler {
 
-    private final Session session;
     private final User user;
     private final Viewer viewer;
 
-    public ClientHandler(Session session, Viewer viewer) {
-        this.session = session;
-        this.user = session.getUser();
-        this.viewer = viewer;
+    public ClientHandler(User user, Viewer viewer) throws NullPointerException {
+        this.user = Objects.requireNonNull(user, "User must not be null");
+        this.viewer = Objects.requireNonNull(viewer, "Viewer must not be null");
     }
 
-    public Message read() throws Exception {
-        return session.readMessage();
-    }
-
-    public void doTask(Message message) throws IOException {
+    public void doTask(Message message, Session session) throws IOException {
```

Comment:
It is always better to have more global parameters first. In that case, the session leaves longer than the message so it is better to have session as the first parameter and the message as the second one